src/eholdings.js

```diff
@@ -0,0 +1,136 @@
+    const { history, location: { search } } = this.props;
+    const searchElement = document.getElementById('eholdings-search');
+
+    if (searchElement) {
+      searchElement.focus();
+    } else {
+      history.push({ pathname: '/eholdings', search });
+    }
+  };
+
+  render() {
+    const {
+      showSettings,
+      match: { path: rootPath },
+    } = this.props;
+
+    return showSettings
+      ? (
+        <CommandList commands={commands}>
+          <Route path={rootPath} component={SettingsRoute}>
+            <Route path={`${rootPath}/knowledge-base/:kbId`} exact component={SettingsKnowledgeBaseRoute} />
+            <Route path={`${rootPath}/:kbId/root-proxy`} exact component={SettingsRootProxyRoute} />
+            <Route path={`${rootPath}/:kbId/custom-labels`} exact component={SettingsCustomLabelsRoute} />
+            <Route path={`${rootPath}/:kbId/access-status-types`} exact component={SettingsAccessStatusTypesRoute} />
+            <Route path={`${rootPath}/:kbId/users`} exact component={SettingsAssignedUsersRoute} />
+            <Route path={`${rootPath}/:kbId/usage-consolidation`} exact component={SettingsUsageConsolidationRoute} />
+          </Route>
+        </CommandList>
+      )
+      : (
+        <>
+          <CommandList commands={commands}>
+            <EHoldingsAppContext />
+            <KeyShortcutsWrapper focusSearchField={this.focusSearchField}>
+              <Route path={rootPath} component={ApplicationRoute}>
+                <Switch>
+                  <Route path={`${rootPath}/providers/:providerId`} exact component={ProviderShow} />
+                  <Route path={`${rootPath}/providers/:providerId/edit`} exact component={ProviderEdit} />
+                  <Route path={`${rootPath}/packages/new`} exact component={PackageCreate} />
+                  <Route path={`${rootPath}/packages/:packageId`} exact component={PackageShow} />
+                  <Route path={`${rootPath}/packages/:packageId/edit`} exact component={PackageEdit} />
+                  <Route path={`${rootPath}/titles/new`} exact component={TitleCreate} />
+                  <Route path={`${rootPath}/titles/:titleId`} exact component={TitleShow} />
+                  <Route path={`${rootPath}/titles/:titleId/edit`} exact component={TitleEdit} />
+                  <Route path={`${rootPath}/resources/:id`} exact component={ResourceShow} />
+                  <Route path={`${rootPath}/resources/:id/edit`} exact component={ResourceEdit} />
+                  <Route path={`${rootPath}/notes/new`} exact component={NoteCreate} />
+                  <Route path={`${rootPath}/notes/:noteId`} exact component={NoteView} />
+                  <Route path={`${rootPath}/notes/:id/edit`} exact component={NoteEdit} />
+                  <Route path={`${rootPath}/`} exact component={SearchRoute} />
+                </Switch>
+              </Route>
+            </KeyShortcutsWrapper>
+          </CommandList>
+        </>
+      );
```

Comment:
We've got two equal wrappers `<CommandList>` for two options. It's better to make some refactoring here.
```suggestion
    return (
      <CommandList commands={commands}>
        {showSettings
          ? (
            <Route path={rootPath} component={SettingsRoute}>
              <Route path={`${rootPath}/knowledge-base/:kbId`} exact component={SettingsKnowledgeBaseRoute} />
              <Route path={`${rootPath}/:kbId/root-proxy`} exact component={SettingsRootProxyRoute} />
              <Route path={`${rootPath}/:kbId/custom-labels`} exact component={SettingsCustomLabelsRoute} />
              <Route path={`${rootPath}/:kbId/access-status-types`} exact component={SettingsAccessStatusTypesRoute} />
              <Route path={`${rootPath}/:kbId/users`} exact component={SettingsAssignedUsersRoute} />
              <Route path={`${rootPath}/:kbId/usage-consolidation`} exact component={SettingsUsageConsolidationRoute} />
            </Route>
          )
         : (
          <>
            <EHoldingsAppContext />
            <KeyShortcutsWrapper focusSearchField={this.focusSearchField}>
              <Route path={rootPath} component={ApplicationRoute}>
                <Switch>
                  <Route path={`${rootPath}/providers/:providerId`} exact component={ProviderShow} />
                  <Route path={`${rootPath}/providers/:providerId/edit`} exact component={ProviderEdit} />
                  <Route path={`${rootPath}/packages/new`} exact component={PackageCreate} />
                  <Route path={`${rootPath}/packages/:packageId`} exact component={PackageShow} />
                  <Route path={`${rootPath}/packages/:packageId/edit`} exact component={PackageEdit} />
                  <Route path={`${rootPath}/titles/new`} exact component={TitleCreate} />
                  <Route path={`${rootPath}/titles/:titleId`} exact component={TitleShow} />
                  <Route path={`${rootPath}/titles/:titleId/edit`} exact component={TitleEdit} />
                  <Route path={`${rootPath}/resources/:id`} exact component={ResourceShow} />
                  <Route path={`${rootPath}/resources/:id/edit`} exact component={ResourceEdit} />
                  <Route path={`${rootPath}/notes/new`} exact component={NoteCreate} />
                  <Route path={`${rootPath}/notes/:noteId`} exact component={NoteView} />
                  <Route path={`${rootPath}/notes/:id/edit`} exact component={NoteEdit} />
                  <Route path={`${rootPath}/`} exact component={SearchRoute} />
                </Switch>
              </Route>
            </KeyShortcutsWrapper>
          </>
        )}
      </CommandList>
    );
```